### PR TITLE
Community recent items using Newt (aka the shim) 

### PR DESCRIPTION
**Pull request by jimfulton**
This replaces #134

### CODE REVIEW

File: karl/models/newtqbe.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import karl.utils
 import newt.db.search
-from newt.qbe import QBE, match, scalar, text_array, fulltext
+from newt.qbe import QBE, match, scalar, prefix, text_array, fulltext
```
Comment (pauleveritt) [inline karl/models/newtqbe.py]: The fulltext import isn't needed.

**Comment by pauleveritt** [comment]
Looks good, I must say, this is fun stuff to read.

**Comment by jimfulton** [comment]
On Sat, Apr 29, 2017 at 7:48 AM, Paul Everitt <notifications@github.com>
wrote:

> The fulltext import isn't needed.
>
True, but it's referenced in some commented lines that I plan to uncomment
when those helpers are needed.  I'll remove the import though.

**Comment by jimfulton** [comment]
On Sat, Apr 29, 2017 at 7:49 AM, Paul Everitt <notifications@github.com>
wrote:

> Looks good, I must say, this is fun stuff to read.
>
Yeah, now that the foundation is in place, future sqlifications should look
like this and be pretty straightforward to grok.

**Comment by pauleveritt** [comment]
I saw the commented line, didn’t realize it was intended for later. Makes sense then to keep the import.

—Paul

**Comment by jimfulton** [comment]
I'll merge this later in the week after the next release has settled. :)

**Comment by jimfulton** [comment]
I looked at this a bit more in hopes of leveraging an index for both selection and sorting.  I learned some things:

- Sorting here is cheap.  I need to let it go. :)

- Using the path index to filter for community is expensive. I should create a community id index.

- The security check is expensive.   In the common case of batching, it should be possible to mitigate this by only checking enough items to fill the batch, but PG is too dumb to do this.  It insists on applying the limit after all filtering. I can work around this using a CTE as an optimization fence.

So with some changes, the sql hot times can be reduced ~90%.  So, for example, the most expensive hot query time for the first 20 results drops from ~5s to ~.2s.

Getting the CTE used in QBE is kind of a stretch. I need to ponder.

**Comment by jimfulton** [comment]
Well, no, we need the total count, so, no way to short-circuit security checks. Wimper.

**Pull request merged**